frontend/src/pages/PipelineDetails.tsx

```diff
@@ -238,41 +238,47 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     this.clearBanner();
     const pipelineId = this.props.match.params[RouteParams.pipelineId];
 
+    let pipeline: ApiPipeline;
+    let templateResponse: ApiGetTemplateResponse;
+
     try {
-      const [pipeline, templateResponse] = await Promise.all([
-        Apis.pipelineServiceApi.getPipeline(pipelineId),
-        Apis.pipelineServiceApi.getTemplate(pipelineId)
-      ]);
+      pipeline = await Apis.pipelineServiceApi.getPipeline(pipelineId);
+    } catch (err) {
+      await this.showPageError('Cannot retrieve pipeline details.', err);
+      logger.error('Cannot retrieve pipeline details.', err);
```

Comment:
Not needed in this PR, but maybe we should create a util function that does these two lines with the two inputs (message and error) since it's always the same.

We can then add a comment explaining why it's useful to have both the logger call and the page error.

Reply:
Perhaps we should allow the user to see the callstack in the error dialog. It can appear as a collapsed area with a toggle.

@ajayalfred in case he has any thoughts here.

Reply:
+1 to consider it

Reply:
Agree. This will be useful. What do you mean by error dialog? Can you add a screenshot?

Reply:
Ok sounds good. Do you need a spec for this?

Side note, we should have Jordan do a scrub of text in all dialogs so the messaging is consistent.